alpha-core/src/main/java/at/ac/tuwien/kr/alpha/core/programs/transformation/aggregates/AggregateOperatorNormalization.java

```diff
@@ -69,6 +69,24 @@ private AggregateOperatorNormalization() {
 
 	private static List<Literal> rewriteAggregateOperator(AggregateLiteral lit) {
 		AggregateAtom atom = lit.getAtom();
+		if (atom.getLowerBoundOperator() == null && atom.getUpperBoundOperator() != null) {
+			ComparisonOperator operator = atom.getUpperBoundOperator();
+			AggregateLiteral flipped; // "flip" right-hand operator and term to left-hand ones.
+			if (operator.equals(ComparisonOperators.EQ) || operator.equals(ComparisonOperators.NE)) {
+				flipped = Atoms.newAggregateAtom(
+					atom.getUpperBoundOperator(), 
+					atom.getUpperBoundTerm(), 
+					atom.getAggregateFunction(), 
+					atom.getAggregateElements()).toLiteral(!lit.isNegated());
+			} else {
+				flipped = Atoms.newAggregateAtom(
+					atom.getUpperBoundOperator().negate(), 
+					atom.getUpperBoundTerm(), 
+					atom.getAggregateFunction(), 
+					atom.getAggregateElements()).toLiteral(!lit.isNegated());
+			}
```

Comment:
The difference in the second argument seems to be the only difference between the two branches. You could make that more explicit by doing:

```suggestion
			var flippedUpperBoundOperator = EQ.equals(operator) || NE.equals(operator) ? atom.getUpperBoundOperator() : atom.getUpperBoundOperator().negate();
                        // "flip" right-hand operator and term to left-hand ones.
			AggregateLiteral flipped = Atoms.newAggregateAtom(
					flippedUpperBoundOperator, 
					atom.getUpperBoundTerm(), 
					atom.getAggregateFunction(),
					atom.getAggregateElements()).toLiteral(!lit.isNegated());
```